webserver/pkgpkr/webservice/recommender_service.py

```diff
@@ -89,24 +89,35 @@ def get_recommendations(self, dependencies):
         for dependency in recommended:
 
             at_split = dependency.split('@')
-            dependency_name = at_split[0].split('/')[-1]
-            dependency_version = at_split[-1]
+            dependency_name = at_split[len(at_split)-2].split('/')[len(at_split)-3]
 
             d = dict()
 
             d['name'] = dependency
 
             # Get average downloads
             res = requests.get(NPM_LAST_MONTH_DOWNLOADS_META_API_URL + f'/{dependency_name}')
-            average_downloads = self.get_average_monthly_donwloads(res.json()['downloads'])
-            # commas as thousands separators
+
+            # If there are download counts for the package
+            if 'downloads' in res.json():
+                average_downloads = self.get_average_monthly_donwloads(res.json()['downloads'])
+            else:
+                average_downloads = 0
+
+            # Commas as thousands separators
             d['average_downloads'] = f'{average_downloads:,}'
 
-            # Get keywords (i.e. categories) and date
+            # Get keywords (i.e. categories)
             res = requests.get(NPM_DEPENDENCY_META_URL + f'/{dependency_name}')
 
             res_json = res.json()
 
+            # Get lastest version
+            if 'dist-tags' in res_json:
+                dependency_version = res_json['dist-tags']['latest']
```

Comment:
The 'latest' dist-tag is not guaranteed to exist, should set `dependency_version` to '0.0.0' if it doesn't.

Reply:
Changed it to 
if 'dist-tags' in res_json and\
                    'latest' in res_json['dist-tags']:
                dependency_version = res_json['dist-tags']['latest']